### PR TITLE
Make all classes inherit from object

### PR DESCRIPTION
**Pull request by slide**
If no base is provided, then object should be an implicit base class.

### CODE REVIEW

**Comment by slozier** [comment]
I actually removed this on purpose since it was causing failures...

**Comment by slozier** [comment]
So it looks like we need to add object to the class bases, but we shouldn't add it before calling the metaclass...

Here's the relevant test case:
```Python
class MyMetaClass(type):
    @classmethod
    def __prepare__(metacls, name, bases):
        assert bases == ()
        return dict()
    def __new__(cls, name, bases, classdict):
        assert bases == ()
        return type.__new__(cls, name, bases, classdict)

class MyClass(metaclass=MyMetaClass):
    pass

assert MyClass.__bases__ == (object,)
```

**Comment by slide** [comment]
I added it because I was seeing some test failures :)

File: Src/IronPython/Runtime/Operations/PythonOps.cs [hunk 1]
```diff
@@ -1344,11 +1344,6 @@ public static partial class PythonOps {
...
             }
 
-            // all classes should inherit from object, so if no base is provided, add object as a base
-            if(bases.Length == 0) {
-                bases = new object[] { DynamicHelpers.GetPythonTypeFromType(typeof(object)) };
-            }
-
             PythonTuple tupleBases = PythonTuple.MakeTuple(bases);
 
             object metaclass;
```
Comment (slozier) [inline Src/IronPython/Runtime/Operations/PythonOps.cs]: Did you verify what happens if the branch below is taken?

**Pull request merged**